lib/game.rb

```diff
@@ -0,0 +1,90 @@
+class Game
+  attr_reader :rolls
+
+  MAX_PINS = 10
+
+  def initialize
+    @rolls = [] 
+    @max_rolls = 20
+  end
+
+  def roll(pins) 
+    check_pin_number
+    check_game_status
+    @rolls << pins
+      if @rolls.count == 19 && (pins == 10) || @rolls.count == 20 && (@rolls[-1] + @rolls[-2]) == 10
+        @max_rolls += 1
+      elsif first_bowl? && (pins == 10)
+        @rolls << "-" unless completed_all_rolls
+      end
+  end
+
+  def check_game_status
+    raise ("Game Over. Ask for your score!") if completed_all_rolls
+  end
+
+  def total_score
+    pins_score + total_bonus_points
+  end
+  
+  private
+
+  def check_pin_number
+    raise ("This is not valid input.") if pins > MAX_PINS
+  end
+
+  def completed_all_rolls
+    @rolls.count == @max_rolls
+  end
+
+  def pins_score 
+    @scores = @rolls - ["-"]
```

Comment:
Why are you introducing another instance variable here? A local variable is alway preferible.